Modules/xSharePoint/DSCResources/MSFT_xSPAppCatalog/MSFT_xSPAppCatalog.psm1

```diff
@@ -0,0 +1,81 @@
+function Get-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Collections.Hashtable])]
+    param
+    (
+        [parameter(Mandatory = $true)] [System.String] $AppCatalogUrl ,
+        [parameter(Mandatory = $true)]  [System.String] $WebApp, 
+        [parameter(Mandatory = $false)] [System.Management.Automation.PSCredential] $InstallAccount
+
+    )
+
+    Write-Verbose -Message "Checking app urls settings"
+
+    $result = Invoke-xSharePointCommand -Credential $InstallAccount -Arguments $PSBoundParameters -ScriptBlock {
+        $params = $args[0]
+        $wa = Get-SPWebApplication $param.WebApp
+        $feature = $wa.Features[[Guid]::Parse("f8bea737-255e-4758-ab82-e34bb46f5828")]
+        if($feature -eq $null ){
+            return $null
+        }
+        $site = Get-SPSite $feature.Properties["__AppCatSiteId"].Value
```

Comment:
Can we handle a site not existing here (e.g. if someone deletes the site collection after setting it as the app catalog)